test/fmf/tests/fdo-bootc.sh

```diff
@@ -0,0 +1,260 @@
+#!/bin/bash
+set -euox pipefail
+
+# FDO test in RHEL-10 Image Mode
+
+# Local variables
+VM_NAME="fdo-disk-image"
+GUEST_ADDRESS=192.168.100.50
+SSH_USER="admin"
+
+# Set up temporary files.
+TEMPDIR=$(mktemp -d)
+
+# SSH setup.
+SSH_OPTIONS=(-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=5)
+SSH_KEY=rhel-edge/key/ostree_key
+EDGE_USER_PASSWORD=foobar
+
+function clean_up {
+    greenprint "🧼 Cleaning up"
+    sudo rm -rf ./{Containerfile,output,config.toml}
+    sudo podman logout registry.stage.redhat.io
+    sudo podman rmi -a -f
+    sudo rm -rf /var/lib/libvirt/images/{fdo-iso.qcow2,install.iso}
+    sudo virsh net-destroy integration && sudo virsh net-undefine integration
+    VM_STATUS=$(sudo virsh list --all | awk '{print $3}' | tail -n 2)
+    if [[ $VM_STATUS == running ]]; then
+        sudo virsh destroy "${VM_NAME}"
+    fi
+    sudo virsh undefine "${VM_NAME}" --nvram
+    # TO DO: Remove tag from quay.io registry
+    # skopeo delete --creds "${QUAY_USERNAME}:${QUAY_PASSWORD}" "${QUAY_REPO_URL}:${QUAY_REPO_TAG}"
+}
+
+# Colorful output.
+function greenprint {
+    echo -e "\033[1;32m${1}\033[0m"
+}
+
+wait_for_ssh_up () {
+    SSH_STATUS=$(sudo ssh "${SSH_OPTIONS[@]}" -i "${SSH_KEY}" "${SSH_USER}@${1}" '/bin/bash -c "echo -n READY"')
+    if [[ $SSH_STATUS == READY ]]; then
+        echo 1
+    else
+        echo 0
+    fi
+}
+
+check_result () {
+    greenprint "🎏 Checking for test result"
+    if [[ $RESULTS == 1 ]]; then
+        greenprint "💚 Success"
+    else
+        greenprint "❌ Failed"
+        clean_up
+        exit 1
+    fi
+}
+
+trap clean_up EXIT
+
+sudo dnf install -y \
+    cargo \
+    openssl \
+    openssl-devel \
+    git \
+    make \
+    systemd \
+    krb5-devel \
+    python3-docutils \
+    gpgme-devel \
+    libassuan-devel \
+    systemd-rpm-macros \
+    rpmdevtools \
+    golang \
+    go-rpm-macros \
+    python3-devel \
+    selinux-policy-devel \
+    device-mapper-devel \
+    podman \
+    qemu-img \
+    qemu-kvm \
+    libvirt-client \
+    libvirt-daemon-kvm \
+    libvirt-daemon \
+    virt-install
+
+# Clone upstream repo and build fdo packages
+git clone https://github.com/fdo-rs/fido-device-onboard-rs.git && cd fido-device-onboard-rs
+make rpm
+sudo dnf -y install rpmbuild/RPMS/x86_64/*
+cd ..
+
+# Manufacturing server setup
+# Clone downstream repo custom branch (fdo simplified installer setup skipping disk encryption)
+git clone -b fdo-man-server-infra https://github.com/mcattamoredhat/rhel-edge.git && cd rhel-edge
```

Comment:
I don't like using personal git repos as place to host code we are using more broadly.

Could we create a specific branch under https://github.com/virt-s1/rhel-edge that could contain these modifications?

Longer term, I also worry about keeping the modified version of the test in sync with the main version. Can we explore a way to not have to use this fork/modified version as a future piece of work?

Reply:
Agreed, ultimately anything forked for the short term should have a route back upstream to reduce ongoing rebasing, but in the short term we could also have a repo in this org if a PR to get it back upstream isn't possible. It would be good to have the details of the options documented in this ticket or somewhere public.

Reply:
Done. We will not use a specific branch outside this repo.